app/src/main/java/com/asfoundation/wallet/interact/SmsValidationInteract.kt

```diff
@@ -20,6 +23,19 @@ class SmsValidationInteract(
   fun validateCode(phoneNumber: String, wallet: Wallet,
                    code: String): Single<WalletValidationStatus> {
     return smsValidationRepository.validateCode(phoneNumber, wallet.address, code)
+        .doOnSuccess { saveWalletVerifiedStatus(it, wallet.address) }
+  }
+
+  private fun saveWalletVerifiedStatus(status: WalletValidationStatus, walletAddress: String) {
+    if (status == WalletValidationStatus.DOUBLE_SPENT || status == WalletValidationStatus.SUCCESS) {
+      defaultSharedPreferences.edit()
+          .putBoolean(WALLET_VERIFIED + walletAddress, true)
+          .apply()
```

Comment:
This logic should be inside the sharedPreference. The only thing that should be here is the if